fix(upgrade): expose upgrade component scope to subclass

smdunn809 opened the pull request:
Expose upgrade component's scope to be accessible to subclass in order to access the scope that UpgradeComponent creates in its constructor. This is helpful if declaring other services that need the same scope as the UpgradeComponent.

Fixes #41756

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: 41756


## What is the new behavior?

When extending UpgradeComponent, it creates a new scope for the upgraded component. In my case, I need to be able to access this scope to ensure any other controllers or components I create are created using this new scope.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x ] No

smdunn809 committed:
fix(upgrade): expose upgrade component scope to subclass

Expose upgrade component's scope to be accessible to subclass in order to access the scope that UpgradeComponent creates in its constructor. This is helpful if declaring other services that need the same scope as the UpgradeComponent.

Fixes #41756

kseamon commented on packages/upgrade/static/src/upgrade_component.ts:
```diff
@@ -74,7 +74,7 @@ export class UpgradeComponent implements OnInit, OnChanges, DoCheck, OnDestroy {
   private element: Element;
   private $element: IAugmentedJQuery;
-  private $componentScope: IScope;
+  protected $componentScope: IScope;
```
Can we also make it readonly?

petebacondarwin replied:
👍

smdunn809 replied:
Sure, updated to be readonly

petebacondarwin commented:
For visibility - I am not understanding the use-case for which this is needed.

https://github.com/angular/angular/issues/41756#issuecomment-825569726

smdunn809 committed:
fixup! fix(upgrade): expose upgrade component scope to subclass

Expose upgrade component's scope to be accessible to subclass in order to access the scope that UpgradeComponent creates in its constructor. This is helpful if declaring other services that need the same scope as the UpgradeComponent.

Fixes #41756

petebacondarwin commented:
I approved, but first, let's add the `readonly` before we land this.

petebacondarwin reviewed (changes requested):
@smdunn809 - thanks for the update. Could you amend the second commit to be a fixup, so that we can autosquash it when we merge? See https://github.com/angular/angular/blob/master/docs/FIXUP_COMMITS.md if you want more info.

AndrewKushnir reviewed (commented):
I noticed that the conversation is in progress in #41756 (so the final solution might be changed), but if we end up making the `$componentScope` protected, I'd like to propose adding a comment in the code to explain why we did that (and why only that field is protected, but not `$injector` for ex.), so there is an additional context for easier maintenance in the future.

Also I think we should change the commit type from `fix(upgrade)` -> `refactor(upgrade)` as it's technically not a bugfix, but more of an improvement for specific use-cases.
